kcalcore/icaltimezones.cpp

```diff
@@ -1360,21 +1358,19 @@ QList<QDateTime> ICalTimeZoneSourcePrivate::parsePhase( icalcomponent *c,
         ICalFormat icf;
         ICalFormatImpl impl( &icf );
         impl.readRecurrence( icalproperty_get_rrule( p ), &r );
-        r.setStartDt( klocalStart );
-        // The end date time specified in an RRULE should be in UTC.
-        // Convert to local time to avoid timesInInterval() getting things wrong.
-        if ( r.duration() == 0 ) {
-          KDateTime end( r.endDt() );
-          if ( end.timeSpec() == KDateTime::Spec::UTC() ) {
-            end.setTimeSpec( KDateTime::Spec::ClockTime() );
-            r.setEndDt( end.addSecs( prevOffset ) );
-          }
+        r.setStartDt( kutcStart );
```

Comment:
Use UTC based timestamps for dt start and the interval boundaries below.
Transformation back to UTC is no longer necessary (lines 1376 & 1377)